common/testutils/test_random.go

```diff
@@ -0,0 +1,108 @@
+package testutils
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+// TestRandom provides all the functionality of math/rand.Rand, plus additional randomness functionality useful for testing
+// This struct wraps an instance of rand.Rand, and directly delegates all standard random methods to this internal instance
+type TestRandom struct {
+	internalRand *rand.Rand
+}
+
+// NewTestRandom creates a new instance of TestRandom
+// This method may either be seeded, or not seeded. If no seed is provided, then current unix nano time is used.
+func NewTestRandom(fixedSeed ...int64) *TestRandom {
+	var seed int64
+	if len(fixedSeed) == 0 {
+		seed = time.Now().UnixNano()
+	} else if len(fixedSeed) == 1 {
+		seed = fixedSeed[0]
+	} else {
+		panic("too many arguments, expected exactly one seed")
+	}
+
+	fmt.Printf("Random seed: %d\n", seed)
+	return &TestRandom{
+		internalRand: rand.New(rand.NewSource(seed)),
+	}
+}
+
+// GetRand returns the underlying random instance
+func (r *TestRandom) GetRand() *rand.Rand {
+	return r.internalRand
+}
+
+// RandomBytes generates a random byte slice of a given length.
+func (r *TestRandom) RandomBytes(length int) []byte {
+	bytes := make([]byte, length)
+	_, err := r.internalRand.Read(bytes)
+	if err != nil {
+		panic(err)
+	}
+	return bytes
+}
+
+// RandomTime generates a random time.
+func (r *TestRandom) RandomTime() time.Time {
+	return time.Unix(int64(r.internalRand.Int31()), 0)
```

Comment:
Nit: any reason we use Int31 instead of Int63?
Also not sure how this function is used but might be good to have a `RandomTimeBetween(start, end time.Time)` version? We can add if/when needed though.

Reply:
No reason, changed to int63

Agreed `RandomTimeBetween` would likely find usage, I'll leave for a future PR though

[09e6390](https://github.com/Layr-Labs/eigenda/pull/976/commits/09e63904bb2ac75517d8e9ff4f0c25ea31cda94d)